src/cgi-newtonraphson.cpp

```diff
@@ -0,0 +1,63 @@
+// this C++ snippet is stored as src/cgi-newtonraphson.hpp
+#include <string>
+#include <iostream>
+#include <nlohmann/json.hpp>
+
+// this C++ code snippet is later referred to as <<algorithm>>
+#include "newtonraphson.hpp"
+
+namespace rootfinding
+{
+
+// An example function is x^3 - x^2  + 2
+double func(double x)
+{
+  return x * x * x - x * x + 2;
+}
+
+// Derivative of the above function which is 3*x^x - 2*x
+double derivFunc(double x)
```

Comment:
```suggestion
double derivative(double x)
```

Reply:
Will be fixed in https://github.com/NLESC-JCER/cpp2wasm/issues/29